File: readability.py
try: input=raw_input
except NameError: pass
passage = input("Text: ")
sentences=0
letters = 0
words=0
for i in passage:
    if i.isalpha():
        letters+=1
    elif i.isspace():
        words+=1
    elif i=='.' or i =='!' or i=='?':
        sentences+=1
words+=1
AverageWords=100.0/words
l=AverageWords*letters
s=AverageWords*sentences
grade=0.0588*l-0.296*s-15.8
if grade>16:
    print(f"Grade 16+")
elif grade<1:
    print(f"Before Grade 1")
else:
    print(f"Grade:{round(grade)}")

File: readability.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(char text[1024]);
int count_words(char text[1024]);
int count_sentences(char text[1024]);

int main(){
    int letters,words,sentences,grade;
    float l,s;
    char text[1024];
    printf("Text: ");
    fgets(text,1024,stdin);
    letters=count_letters(text);
    words=count_words(text);
    sentences=count_sentences(text);
    l=100.0/words*letters;
    s=100.0/words*sentences;
    grade=round(0.0588*l-0.296*s-15.8);
    if(grade>16){
        printf("Grade 16+\n");
    }else if (grade<1)
    {
        printf("Before Grade 1\n");
    }else{
        printf("Grade %d\n",grade);
    }
}

int count_letters(char text[1024]){
    int count,i;
    count=0;
    for(i=0;text[i]!=0;i++){
        if(isalpha(text[i])){
            count++;
        }
    }return count;
}
int count_words(char text[1024]){
    int count;
    count=0;
    for(int i=0;text[i]!=0;i++){
        if(isspace(text[i]))
        {
        count++;
        }
    }return count;
}
int count_sentences(char text[1024]){
    int count;
    count=0;
    for (int i = 0; text[i]!=0; i++)
    {
        if(text[i]=='.' || text[i]=='!' || text[i]=='?'){
            count++;
        }
    }
    return count;
}
